src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -542,12 +544,14 @@ def add_properties(instance, argument_values, force_string):
                 try:
                     argument = shell_safe_json_parse(argument)
                 except (ValueError, CLIError):
-                    pass
+                    json_parse_fail_args.append(argument)
```

Comment:
`shell_safe_json_parse` already has the exception handling logic.

https://github.com/Azure/azure-cli/blob/f34ca47e441c949c75df07d3bfa8e76e4f5d7a28/src/azure-cli-core/azure/cli/core/util.py#L513-L531

Reply:
--add option allows users to pass a string, not necessarily a JSON string. So warning will not show until DeserializationError is thrown. As your advice, the error raised by `shell_safe_json_parse ` has been reserved for generating warning message in latest code version